testing-guidance/workflow-tests/ragdoll/ragdoll-authoring-improvements.md

```diff
@@ -0,0 +1,47 @@
+# Ragdoll Authoring Improvements Workflow Tests
+
+Testing in this area should focus on the new Ragdoll Authoring Improvements.
+
+## General Docs
+* [Creating and Simulating a PhysX Ragdoll](https://www.o3de.org/docs/user-guide/visualization/animation/animation-editor/creating-and-simulating-physx-ragdoll/)
+* [Animation Editor User Interface](https://www.o3de.org/docs/user-guide/visualization/animation/animation-editor/user-interface/)
+* [Setting up a Ragdoll](https://stabilization-22-10--o3deorg.netlify.app/docs/user-guide/visualization/animation/animation-editor/ragdoll/ragdoll-physics-setup/)
+
+## Common Issues to Watch For
+
+Test guidance will sometimes note specific issues to watch for. The common issues below should be watched for through all testing, even if unrelated to the current workflow being tested.
+- Issues with the Editor running in the background when Animation Editor is running (for example freezing when saving changes to an actor).
+- Wiremesh highlighting/coloration in the Atom Render Window does not change properly when hovered over, or is not updated properly when modified.
+- Certain actions not being written to the Action History, and being unaffected by the Undo/Redo actions.
+- No warning/save popups in the Animation Editor (for example when an Autosave is performed, or when a Motion Set is mising when attempting to use "Automatic joint limit setup").
+
+## Workflows
+
+### Area: Ragdoll Authoring Improvements
+
+**Project Requirements**
+
+Any project that has the following Gems enabled:
+* Motion Matching (Required Rin.fbx actor)
+
+**Editor Platforms:**
+* Windows
+* Linux
+
+**Product:** 
+
+**Suggested Time Box:** 45 minutes per platform.
+
+| Workflow                                  | Requests           | Things to Watch For |
```

Comment:
Let's add another workflow where a configured Ragdoll is added to a simple scene and triggered to ragdoll due to an event. Otherwise we've tested that we can set up a ragdoll using the tool, but have not actually tested that the ragdoll generated from the tool works. This workflow can be described in minimal generic form, and allow for creativity from there.

This can be added below the "Picking and Hovering" workflow.

Reply:
Due to an outdated documentation page (https://github.com/o3de/o3de.org/issues/1766) I was unable to prepare this workflow in its entirety. I have managed to make the Rin.fbx actor run for a few seconds and activate ragdoll; however, ragdoll activation causes the actor to flail around. I feel like I am missing some detail about the entity/actor setup/skeleton setup; do you know what could be causing this result (please refer to the attached video)?

For now I will push the changes made to the original file I made yesterday. Other than that, let me know how you would like me to handle adding the "Trigger ragdoll in a scene" workflow.

https://user-images.githubusercontent.com/86952082/191258904-6bd927a1-2f22-4969-a30a-c1c02e764a30.mp4

Reply:
@moraaar @greerdv @amzn-tommy 
Do we have any references that @LB-ArturZieba can use to finalize this workflow, or is the above another bug that should be filed?

@LB-ArturZieba In the meantime, let's go ahead and create the workflow with what you have currently, and we can update the final workflow as that information comes in, docs are updated, etc.

Reply:
Created the basic workflow area for the "Ragdoll in the scene" as requested, and pushed relevant commit.

Reply:
Hi @LB-ArturZieba, do you have a character controller on your entity? And are the collision layers set up to not collide with ragdoll colliders? That is a problematic part of our current workflow which we want to improve.

Otherwise, could you send a physics visual debugger capture please?

Reply:
I also wonder if it would make sense to make authoring ragdolls and simulating them into separate workflows?

Reply:
@greerdv Currently there's two workflow areas in this doc. One is the "Authoring" and the other is the "Simulating" in the editor.

Keeping everything in one workflow file will help somebody that's trying to test/work holistically around Ragdolls get a better idea of what's going on instead of digging through multiple files. Breaking things into different files has it's advantages as well, however I'm of the opinion of keeping them together in a single file.

Reply:
Hi @greerdv, there is no PhysX Character Controller on the entity, the only present components are: Actor, Anim Graph and PhysX Ragdoll. I have also attempted using PhysX Character Collider and PhysX Character Gameplay components, as well as creating separate Collision Group/Layers for them, but results were the same. Attaching a video with a side-by-side comparison with the PhysX Visual Debugger open:

https://user-images.githubusercontent.com/86952082/191688752-d9aa4d06-d906-4f2e-a331-c039cf1f6829.mp4

Reply:
Could you save the capture from the PVD and send the .pxd2 file please?

Reply:
Here is the .pxd2 file (GitHub does not allow for .pxd2 extension file to be uploaded, so I packed it in .zip archive):

[RagdollPVD.zip](https://github.com/o3de/sig-simulation/files/9624413/RagdollPVD.zip)

Reply:
There should be an option to enable mass ratio clamping in the ragdoll component, can you try turning that on please?

Reply:
Sorry for taking long to answer this, as I was unavailable - I have reverified the ragdoll behavior after enabling the "Enable Mass Ratio Clamping" property (left the "Maximum Mass Ratio" at default value = 2.0), and it stopped jittering. Added a relevant note in a new PR regarding this: https://github.com/o3de/sig-simulation/pull/25

Thanks for the help, and here is a short video showing the change:

https://user-images.githubusercontent.com/86952082/192460873-21ed3d1c-ca16-4f21-9b94-0f3ca16c9f79.mp4

Reply:
OK, thanks for verifying. I'll look into revisiting that setting (e.g. turning it on by default)

Reply:
I've made https://github.com/o3de/o3de/issues/12267 to look into improvements around that mass ratio issue.